ocaml/xapi/message_forwarding.ml

```diff
@@ -3921,14 +3925,22 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 		let attach ~__context ~self =
 			info "Pool_update.attach: pool update = '%s'" (pool_update_uuid ~__context self);
 			let local_fn = Local.Pool_update.attach ~self in
-			VDI.forward_vdi_op ~local_fn ~__context ~self:(Db.Pool_update.get_vdi ~__context ~self)
+			let update_vdi = Db.Pool_update.get_vdi ~__context ~self in
+			if Db.is_valid_ref __context update_vdi then
+			VDI.forward_vdi_op ~local_fn ~__context ~self:update_vdi
 				(fun session_id rpc -> Client.Pool_update.attach rpc session_id self)
+			else
+			    raise (Api_errors.Server_error(Api_errors.cannot_find_update, [(pool_update_uuid ~__context self)]))
 
 		let detach ~__context ~self =
 			info "Pool_update.detach: pool update = '%s''" (pool_update_uuid ~__context self);
 			let local_fn = Local.Pool_update.detach ~self in
-			VDI.forward_vdi_op ~local_fn ~__context ~self:(Db.Pool_update.get_vdi ~__context ~self)
+			let update_vdi = Db.Pool_update.get_vdi ~__context ~self in
+			if Db.is_valid_ref __context update_vdi then
+			VDI.forward_vdi_op ~local_fn ~__context ~self:update_vdi
```

Comment:
Indentation again: this `VDI.forward_vdi_op...` line should be more indented than the `if` above it.